bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java

```diff
@@ -277,19 +314,47 @@ public BigtableAsyncAdmin(BigtableAsyncConnection asyncConnection) throws IOExce
   // ****** TO BE IMPLEMENTED [start] ******
 
   @Override
-  public CompletableFuture<Void> deleteColumnFamily(TableName arg0, byte[] arg1) {
-    throw new UnsupportedOperationException("deleteColumnFamily"); // TODO
+  public CompletableFuture<Void> deleteColumnFamily(TableName tableName, byte[] columnName) {
+    return CompletableFuture.supplyAsync(() -> {
+      Modification.Builder modification = Modification.newBuilder().setId(Bytes.toString(columnName))
+          .setDrop(true);
+      ModifyColumnFamiliesRequest.Builder modifyColumnBuilder = ModifyColumnFamiliesRequest
+          .newBuilder().addModifications(modification).setName(toBigtableName(tableName));
+      return modifyColumnBuilder;
+    }).thenCompose(
+        d -> FutureUtils.toCompletableFuture(
+            bigtableTableAdminClient.modifyColumnFamilyAsync(d.build()))
+            .thenApply(r -> null));
   }
 
   @Override
-  public CompletableFuture<Void> deleteSnapshot(String arg0) {
-    throw new UnsupportedOperationException("deleteSnapshot"); // TODO
-
+  public CompletableFuture<Void> deleteSnapshot(String snapshotName) {
+    return CompletableFuture.supplyAsync(() -> {
+    String btSnapshotName = null;
+    try {
+      btSnapshotName = getClusterName().toSnapshotName(snapshotName);
+    } catch (IOException e) {
+      FutureUtils.failedFuture(e);
```

Comment:
I think that if you return `FutureUtils.failedFuture(e)` then you don't have to do `supplyAsync`